src/server/node_manager.c

```diff
@@ -1089,7 +1089,13 @@ get_vnode_state_op(enum vnode_state_op op)
 static void
 shallow_vnode_free(struct pbsnode *vnode)
 {
+	int i;
+
 	if (vnode) {
+		free(vnode->nd_moms);
+		for (i = 0; i < ND_ATR_LAST; i++) {
+			node_attr_def[i].at_free(&vnode->nd_attr[i]);
+		}
 		free(vnode);
```

Comment:
this seems like a pretty thorough clean, do we still want to call this function "shallow" vnode free ?

Reply:
I've reused `free_pnode` and removed `shallow_vnode_dup`. Please have a look.